### PR TITLE
added descriptions to params and comments to resources in YAML files

### PR DESCRIPTION
**Pull request by markllama**
Added description attributes to parameters and descriptive comments to all resources in the primary YAML files defining the stack.

Each parameter must have a description which indicates the use and effect of the parameter values.

### CODE REVIEW

**Comment by markllama** [comment]
Rebased with OpenStack parameters
Added appropriate comments for SoftwareConfig and SoftwareDeployment on _tune_ansible_

**Comment by markllama** [comment]
Added better comments for cloud-init wait condition resources

File: infra.yaml [hunk 1]
```diff
@@ -8,163 +8,252 @@ description: >
...
 
   ldap_hostname:
+    description: >
+      The hostname of the LDAP server providing OpenShift user information
     type: string
     default: ''
 
   deploy_registry:
+    description: >
+      Enable or disable the internal Docker registry
```
Comment (lebauce) [inline infra.yaml]: I think it's more "Enable or disable automatic provisioning of the Docker registry"

File: infra.yaml [hunk 2]
```diff
@@ -8,163 +8,252 @@ description: >
...
     type: boolean
 
   deploy_router:
+    description: >
+      Enable or disable direct access to the nodes (??)
     type: boolean
```
Comment (lebauce) [inline infra.yaml:202]: "Enable or disable deployment of an Openshift router"

File: infra.yaml [hunk 3]
```diff
@@ -8,163 +8,252 @@ description: >
...
 
+  # OpenShift installer (ansible) parameters
   skip_ansible:
+    description: >
+      Disable host based installation/configuration
     type: boolean
 
   openshift_ansible_git_url:
+    description: >
+      The location of the OpenShift Ansible playbooks
```
Comment (lebauce) [inline infra.yaml]: Maybe we should make it clear that an URL of a Git repository is required

File: infra.yaml [hunk 4]
```diff
@@ -183,9 +272,13 @@ resources:
       user_data_format: SOFTWARE_CONFIG
       user_data: {get_resource: init}
 
+
+  # Ensure that the required software repositories are available
```
Comment (lebauce) [inline infra.yaml]: and that os-{collect,refresh,apply}-config are installed

File: loadbalancer.yaml [hunk 5]
```diff
@@ -178,20 +216,26 @@ resources:
             $WC_NOTIFY: { get_attr: ['wait_handle', 'curl_cli'] }
           template: {get_file: fragments/lb-boot.sh}
 
+  # syncronization
```
Comment (lebauce) [inline loadbalancer.yaml]: nit: synchronization

File: master.yaml [hunk 6]
```diff
@@ -262,12 +336,14 @@ resources:
             $POOL_ID: {get_param: rhn_pool}
           template: {get_file: fragments/rhn-register.sh}
 
+  # Insure that the host software is current
```
Comment (lebauce) [inline master.yaml]: nit: Ensure

File: node.yaml [hunk 7]
```diff
@@ -376,8 +440,11 @@ outputs:
     description: IP address of the node
     value: {get_attr: [floating_ip, floating_ip_address]}
   wc_data:
+    description: Syncronization data
```
Comment (lebauce) [inline node.yaml]: nit: Synchronization

**Pull request merged**